var/spack/repos/builtin/packages/openspeedshop/package.py

```diff
@@ -130,84 +139,220 @@ class Openspeedshop(Package):
 
         MPIOptions = []
 
         # openmpi
         if '+openmpi' in spec:
-            MPIOptions.extend([
-                '-DOPENMPI_DIR=%s' % spec['openmpi'].prefix
-            ])
+            MPIOptions.extend(['-DOPENMPI_DIR=%s' % spec['openmpi'].prefix])
         # mpich
         if '+mpich' in spec:
-            MPIOptions.extend([
-                '-DMPICH_DIR=%s' % spec['mpich'].prefix
-            ])
+            MPIOptions.extend(['-DMPICH_DIR=%s' % spec['mpich'].prefix])
         # mpich2
         if '+mpich2' in spec:
-            MPIOptions.extend([
-                '-DMPICH2_DIR=%s' % spec['mpich2'].prefix
-            ])
+            MPIOptions.extend(['-DMPICH2_DIR=%s' % spec['mpich2'].prefix])
         # mvapich
         if '+mvapich' in spec:
-            MPIOptions.extend([
-                '-DMVAPICH_DIR=%s' % spec['mvapich'].prefix
-            ])
+            MPIOptions.extend(['-DMVAPICH_DIR=%s' % spec['mvapich'].prefix])
         # mvapich2
         if '+mvapich2' in spec:
-            MPIOptions.extend([
-                '-DMVAPICH2_DIR=%s' % spec['mvapich2'].prefix
-            ])
+            MPIOptions.extend(['-DMVAPICH2_DIR=%s' % spec['mvapich2'].prefix])
         # mpt
         if '+mpt' in spec:
-            MPIOptions.extend([
-                '-DMPT_DIR=%s' % spec['mpt'].prefix
-            ])
+            MPIOptions.extend(['-DMPT_DIR=%s' % spec['mpt'].prefix])
 
         cmakeOptions.extend(MPIOptions)
 
-    def install(self, spec, prefix):
-
-        # openmpi_prefix_path = "/opt/openmpi-1.8.2"
-        # mvapich_prefix_path = "/usr/local/tools/mvapich-gnu"
-        #    '-DOPENMPI_DIR=%s'            % spec['openmpi'].prefix,
-        #    '-DOPENMPI_DIR=%s'            % openmpi_prefix_path,
-        #    '-DMVAPICH_DIR=%s'            % mvapich_prefix_path,
+    def setup_environment(self, spack_env, run_env):
+        """Set up the compile and runtime environments for a package."""
+
+        # Common settings to both offline and cbtf versions
+        # of OpenSpeedShop
+        run_env.prepend_path('PATH', self.prefix.bin)
+
+        # sqlite3 path
+        run_env.prepend_path('PATH', self.spec['sqlite'].prefix.bin)
+
+        # python path
+        run_env.prepend_path('PATH', self.spec['python'].prefix.bin)
+
+        # Find Dyninst library path
+        dyninst_libdir = find_libraries(['libdyninstAPI_RT'],
+                                        root=self.spec['dyninst'].prefix,
+                                        shared=True, recurse=True)
+
+        # Set Dyninst RT library path to support OSS loop resolution code
+        run_env.set('DYNINSTAPI_RT_LIB', dyninst_libdir)
+
+        # Find openspeedshop library path
+        oss_libdir = find_libraries(['libopenss-framework'],
+                                    root=self.spec['openspeedshop'].prefix,
+                                    shared=True, recurse=True)
+        run_env.prepend_path('LD_LIBRARY_PATH',
```

Comment:
I was trying to match the current openspeedshop module file which accounts for all the packages used by openspeedshop in its runtime environment.